Stripe/STPLocalizationUtils.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  STPLocalizationUtils.swift
+//  Stripe
+//
+//  Created by Brian Dorfman on 8/11/16.
+//  Copyright © 2016 Stripe, Inc. All rights reserved.
+//
+
+import Foundation
+
+class STPLocalizationUtils: NSObject {
+  /// Acts like NSLocalizedString but tries to find the string in the Stripe
+  /// bundle first if possible.
+  /// If the main app has a localization that we do not support, we want to switch
+  /// to pulling strings from the main bundle instead of our own bundle so that
+  /// users can add translations for our strings without having to fork the sdk.
+  /// At launch, NSBundles' store what language(s) the user requests that they
+  /// actually have translations for in `preferredLocalizations`.
+  /// We compare our framework's resource bundle to the main app's bundle, and
+  /// if their language choice doesn't match up we switch to pulling strings
+  /// from the main bundle instead.
+  /// This also prevents language mismatches. E.g. the user lists portuguese and
+  /// then spanish as their preferred languages. The main app supports both so all its
+  /// strings are in pt, but we support spanish so our bundle marks es as our
+  /// preferred language and our strings are in es.
+
+  static let localizedStripeStringUseMainBundle: Bool = {
+    if STPBundleLocator.stripeResourcesBundle.preferredLocalizations.first
+      != Bundle.main.preferredLocalizations.first
+    {
+      return true
+    }
+    return false
+  }()
+
+  class func localizedStripeString(forKey key: String) -> String {
+    if languageOverride != nil {
```

Comment:
this languageOverride stuff is imntentionally new?

Reply:
It used to be in a private subclass for localization testing, but we can't override `localizedStripeString` from the tests in Swift. We could do something with dynamic dispatch, but this felt less hacky